--- openssl-0.9.8h/crypto/ossl_typ.h.orig	2008-09-09 17:31:09.000000000 -0700
+++ openssl-0.9.8h/crypto/ossl_typ.h	2008-09-09 17:31:48.000000000 -0700
@@ -172,8 +172,4 @@
 typedef int CRYPTO_EX_dup(CRYPTO_EX_DATA *to, CRYPTO_EX_DATA *from, void *from_d, 
 					int idx, long argl, void *argp);
 
-typedef struct ocsp_req_ctx_st OCSP_REQ_CTX;
-typedef struct ocsp_response_st OCSP_RESPONSE;
-typedef struct ocsp_responder_id_st OCSP_RESPID;
-
 #endif /* def HEADER_OPENSSL_TYPES_H */
 
--- openssl-0.9.8h/crypto/ocsp/ocsp.h.orig	2008-09-09 17:31:25.000000000 -0700
+++ openssl-0.9.8h/crypto/ocsp/ocsp.h	2008-09-09 17:32:12.000000000 -0700
@@ -72,6 +72,11 @@
 extern "C" {
 #endif
 
+
+typedef struct ocsp_req_ctx_st OCSP_REQ_CTX;
+typedef struct ocsp_response_st OCSP_RESPONSE;
+typedef struct ocsp_responder_id_st OCSP_RESPID;
+
 /* Various flags and values */
 
 #define OCSP_DEFAULT_NONCE_LENGTH	16
