--- yajl-1.0.4/CMakeLists.txt.orig	2009-05-13 15:45:08.000000000 -0700
+++ yajl-1.0.4/CMakeLists.txt	2009-05-13 15:46:31.000000000 -0700
@@ -61,10 +61,18 @@
   SET(CMAKE_C_FLAGS_RELEASE "/D NDEBUG /O2")
 ELSE (WIN32)
   SET(CMAKE_C_FLAGS
-      "${CMAKE_C_FLAGS} -fPIC -ansi -pedantic -Wpointer-arith -Wno-format-y2k -Wstrict-prototypes -Wmissing-declarations -Wnested-externs -Wextra	-Wundef -Wwrite-strings -Wold-style-definition -Wno-missing-field-initializers -Wredundant-decls -Wno-unused-parameter -Wno-sign-compare -Wmissing-prototypes")
+      "${CMAKE_C_FLAGS} -fPIC -ansi -pedantic -Wpointer-arith -Wno-format-y2k -Wstrict-prototypes -Wmissing-declarations -Wnested-externs -Wextra	-Wundef -Wwrite-strings -Wold-style-definition -Wno-missing-field-initializers -Wredundant-decls -Wno-unused-parameter -Wno-sign-compare -Wmissing-prototypes -isysroot /Developer/SDKs/MacOSX10.4u.sdk -arch i386 -mmacosx-version-min=10.4")
 
   SET(CMAKE_C_FLAGS_DEBUG "-DDEBUG -g")
   SET(CMAKE_C_FLAGS_RELEASE "-DNDEBUG -O2 -Wuninitialized")
+
+  SET(linkFlags "-mmacosx-version-min=10.4 -isysroot /Developer/SDKs/MacOSX10.4u.sdk -syslibroot,/Developer/SDKs/MacOSX10.4.u.sdk -arch i386")
+  SET(CMAKE_EXE_LINKER_FLAGS    "${linkFlags}"
+      CACHE STRING "YAJL linker flags" FORCE)
+  SET(CMAKE_SHARED_LINKER_FLAGS    "${linkFlags}"
+      CACHE STRING "YAJL linker flags" FORCE)
+  SET(CMAKE_MODULE_LINKER_FLAGS    "${linkFlags}"
+      CACHE STRING "YAJL linker flags" FORCE)
 ENDIF (WIN32) 
 
 
